modules/services/views/src/main/java/au/com/shiftyjelly/pocketcasts/views/helper/SwipeButtonLayoutViewModel.kt

```diff
@@ -0,0 +1,259 @@
+package au.com.shiftyjelly.pocketcasts.views.helper
+
+import android.content.Context
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import au.com.shiftyjelly.pocketcasts.analytics.AnalyticsEvent
+import au.com.shiftyjelly.pocketcasts.analytics.AnalyticsTrackerWrapper
+import au.com.shiftyjelly.pocketcasts.analytics.EpisodeAnalytics
+import au.com.shiftyjelly.pocketcasts.analytics.SourceView
+import au.com.shiftyjelly.pocketcasts.models.entity.BaseEpisode
+import au.com.shiftyjelly.pocketcasts.models.entity.PodcastEpisode
+import au.com.shiftyjelly.pocketcasts.models.entity.UserEpisode
+import au.com.shiftyjelly.pocketcasts.preferences.Settings
+import au.com.shiftyjelly.pocketcasts.repositories.playback.PlaybackManager
+import au.com.shiftyjelly.pocketcasts.repositories.podcast.EpisodeManager
+import au.com.shiftyjelly.pocketcasts.repositories.podcast.PodcastManager
+import au.com.shiftyjelly.pocketcasts.repositories.podcast.UserEpisodeManager
+import au.com.shiftyjelly.pocketcasts.views.dialog.ShareDialog
+import dagger.hilt.android.lifecycle.HiltViewModel
+import dagger.hilt.android.qualifiers.ApplicationContext
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class SwipeButtonLayoutViewModel @Inject constructor(
+    private val analyticsTracker: AnalyticsTrackerWrapper,
+    @ApplicationContext private val context: Context,
+    private val episodeAnalytics: EpisodeAnalytics,
+    private val episodeManager: EpisodeManager,
+    private val playbackManager: PlaybackManager,
+    private val podcastManager: PodcastManager,
+    private val userEpisodeManager: UserEpisodeManager,
+) : ViewModel() {
+
+    fun share(
+        episode: PodcastEpisode,
+        fragmentManager: FragmentManager,
+        context: Context,
+        swipeSource: EpisodeItemTouchHelper.SwipeSource
+    ) {
+
+        viewModelScope.launch(Dispatchers.Default) {
+
+            trackSwipeAction(
+                swipeSource = swipeSource,
+                swipeAction = EpisodeItemTouchHelper.SwipeAction.SHARE,
+            )
+
+            val podcast = podcastManager.findPodcastByUuid(episode.podcastUuid) ?: return@launch
```

Comment:
```suggestion
        viewModelScope.launch {

            trackSwipeAction(
                swipeSource = swipeSource,
                swipeAction = EpisodeItemTouchHelper.SwipeAction.SHARE,
            )

            val podcast = podcastManager.findPodcastByUuidSuspend(episode.podcastUuid) ?: return@launch
```

Should we use the coroutine version `findPodcastByUuidSuspend` and let Room decide the dispatcher it wants to use?

Reply:
Yes, this is much better. Thanks!